### ISSUE TITLE
Adding generic commands

### ISSUE THREAD

**Issue by nasir-awan** [kerryjiang/SuperSocket#542]
I tried to register commands that is generic one, getting error, 

 **_Duplicated command with Key Ping`1 is found_**

```
`public class Ping<T> : IAsyncCommand<ChatSession, PackageInfo>
    {
        public ValueTask ExecuteAsync(ChatSession session, PackageInfo package)
        {
            throw new NotImplementedException();
        }
    }`
```

//Registering commands

 ```
var webSocketHostBuilder = WebSocketHostBuilder.Create(builder);
            webSocketHostBuilder                 
                .UseSession<ChatSession>()
                .UseCommand<PackageInfo, StringPackageConverter>((commandOptions) =>                
                {
                    commandOptions.AddCommand<Ping<TestClass1>>();
                    commandOptions.AddCommand<Ping<TestClass2>>();
                })
```

is there anyway to do it?

**Comment by nasir-awan** [comment]
any update regarding this issue?

**Comment by ScarletKuro** [comment]
No, there is currently no way to do it, because I'm not sure how you want SuperSocket to understand what command it should invoke.
What you can try in theory is to make an possible workaround, something like this:
```
public abstract class PingBase<T> : IAsyncCommand<ChatSession, PackageInfo>
{
...
}
```
and then have separate pings 
```
[Command(Key = "Ping1")]
public class Ping1 : PingBase<TestClass1>
{
}

[Command(Key = "Ping2")]
public class  Ping2 : PingBase<TestClass2>
{
}
```
and in your `IPackageDecoder<PacketInfo>` while decoding the package you specify the Key what command it should trigger

**Comment by ScarletKuro** [comment]
I also thought about another way. You can ditch the build in Commands and implement your own IPackageHandler<PacketInfo> class and add tell SuperSocket to use it instead, then you can do whatever you want and decide how and what class you want to use to invoke. This is something I personally use.

**Comment by nasir-awan** [comment]
> No, there is currently no way to do it, because I'm not sure how you want SuperSocket to understand what command it should invoke. What you can try in theory is to make an possible workaround, something like this:
> 
> ```
> public abstract class PingBase<T> : IAsyncCommand<ChatSession, PackageInfo>
> {
> ...
> }
> ```
> 
> and then have separate pings
> 
> ```
> [Command(Key = "Ping1")]
> public class Ping1 : PingBase<TestClass1>
> {
> }
> 
> [Command(Key = "Ping2")]
> public class  Ping2 : PingBase<TestClass2>
> {
> }
> 
> .UseCommand<PackageInfo, StringPackageConverter>((commandOptions) =>                
>                {
>                    commandOptions.AddCommand<Ping1>();
>                    commandOptions.AddCommand<Ping2>();
>                })
> ```
> 
> and in your `IPackageDecoder<PacketInfo>` while decoding the package you specify the Key what command it should trigger

@ScarletKuro  Thanks for you reply

yes, that could also possible 
If we have large numbers of commands, need to write a wrapper class each of them.
